src/sourmash/tax/tax_utils.py

```diff
@@ -36,7 +39,272 @@
 
 # import lca utils as needed for now
 from sourmash.lca import lca_utils
-from sourmash.lca.lca_utils import (LineagePair, taxlist, display_lineage, pop_to_rank)
+from sourmash.lca.lca_utils import (taxlist, display_lineage, pop_to_rank)
+
+class LineagePair(NamedTuple):
+    rank: str
+    name: str = None
+    taxid: int = None
+
+@dataclass(frozen=True, order=True)
+class BaseLineageInfo:
+    """BaseLineageInfo Class, defining a set of functions that can be used to handle
+       hierarchical set of LineagePairs. """
+    # need to set compare=False for any mutable type to keep this class hashable
+    ranks: tuple() # require ranks
+    lineage: tuple = field(default=()) #tuple of LineagePairs
+    lineage_str: str = field(default=None, compare=False) # ';'- or ','-separated str of lineage names
```

Comment:
return to later: why these two (`lineage_str` and `lineage_dict`)?

Reply:
- `lineage_str` is the easiest to use to initialize (RankLineageInfo(lineage_str= "a;b;c"))
- `lineage_dict` makes it simple to initialize from a tax csv row, which typically has a column per rank, e.g. `{"superkingdom": "a", "phylum": "b"}`

Reply:
I think autodetection is worse, because it's not always obvious what will work.

Hear, maybe just add the info above to the comments?  It wasn' clear to me that these were primarily for use in initialization, which then makes it all make sense!

Reply:
additional info added in https://github.com/sourmash-bio/sourmash/pull/2437/commits/48af504a055f4b675bc94da434cc3dec724961dd